src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.Helpers.cs

```diff
@@ -23,7 +25,21 @@ private static void ValidateInputOutputSpanNonOverlapping<T>(ReadOnlySpan<T> inp
         }
 
         /// <summary>Throws an <see cref="OverflowException"/> for trying to negate the minimum value of a two-complement value.</summary>
-        internal static void ThrowNegateTwosCompOverflow() => throw new OverflowException(SR.Overflow_NegateTwosCompNum);
+        private static void ThrowNegateTwosCompOverflow() => throw new OverflowException(SR.Overflow_NegateTwosCompNum);
+
+        /// <summary>Creates a span of <typeparamref name="TTo"/> from a <typeparamref name="TTo"/> when they're the same type.</summary>
+        private static unsafe Span<TTo> Rename<TFrom, TTo>(Span<TFrom> span) // MemoryMarshal.Cast, but without constraints or validation
```

Comment:
```suggestion
        private static unsafe Span<TTo> CastUnsafe<TFrom, TTo>(Span<TFrom> span) // MemoryMarshal.Cast, but without constraints or validation
```

Reply:
It's not an arbitrary cast, though: TFrom and TTo are the same type.

Reply:
`CastSameSize`?

Something about `Rename` just doesn't sit right.

Reply:
It's not even about being the same size, it's about literally being the same type. It's used where we're calling it T, we validate that T == char, and then we want to call it char, hence why I chose rename. I don't think it really matters; if we can come up with something better and similarly concise in the future, we can rename it then (no pun intended).